yb-voyager/cmd/export.go

```diff
@@ -417,3 +428,44 @@ func checkDependenciesForExport() (binaryCheckIssues []string, err error) {
 
 	return binaryCheckIssues, nil
 }
+
+func checkJavaVersion() (binaryCheckIssue string, err error) {
+	javaBinary := "java"
+	if javaHome := os.Getenv("JAVA_HOME"); javaHome != "" {
+		javaBinary = javaHome + "/bin/java"
+	}
+
+	cmd := exec.Command(javaBinary, "-version")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Sprintf("java: required version >= %d", MIN_REQUIRED_JAVA_VERSION), nil
+	}
+
+	// Example output
+	// openjdk version "17" 2021-09-14
+	// OpenJDK Runtime Environment (build 17+35-2724)
+	// OpenJDK 64-Bit Server VM (build 17+35-2724, mixed mode, sharing)
+	versionOutput := string(output)
```

Comment:
pls test this with a few java versions on all OS (mac, ubuntu, centos) just to make sure our assumptions here(version is the 3rd part, etc) are valid on all OS

Reply:
I have now modified this to basically mimic the steps we had in run.sh to extract the version.
We search for a line that contains the string 'version' and get that and over there we have the actual java version enclosed in ""

Reply:
Got it, would still recommend doing a sanity test on ubuntu, rhel atleast. Because failing voyager because we weren't able to properly figure out the version would be bad. (especially since we don't even have a prompt for these dependency checks).

Reply:
Yeah I checked in on ubuntu and rhel8

Reply:
And macos too